railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -50,10 +50,10 @@ def gitignore
     end
 
     def lib
-      template "lib/%name%.rb"
-      template "lib/tasks/%name%_tasks.rake"
-      template "lib/%name%/version.rb"
-      template "lib/%name%/engine.rb" if engine?
+      template "lib/%underscored_name%.rb"
+      template "lib/tasks/%namespaced_name%_tasks.rake"
```

Comment:
This is already correct, isn't it? `foo-bar` will generate: `lib/tasks/foo/bar_tasks.rake`